client/state/posts/reducer.js

```diff
@@ -163,40 +170,33 @@ export const queries = ( () => {
 
 	return createReducer( {}, {
 		[ POST_RESTORE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: '__RESTORE_PENDING'
 			}, { patch: true } );
 		},
 		[ POST_RESTORE_FAILURE ]: ( state, { siteId, postId } ) => {
-			return applyToManager( state, siteId, 'receive', {
+			return applyToManager( state, siteId, 'receive', false, {
 				ID: postId,
 				status: 'trash'
 			}, { patch: true } );
 		},
 		[ POSTS_REQUEST_SUCCESS ]: ( state, { siteId, query, posts, found } ) => {
-			if ( ! state[ siteId ] ) {
-				state = {
-					...state,
-					[ siteId ]: new PostQueryManager()
-				};
-			}
-
-			return applyToManager( state, siteId, 'receive', posts, { query, found } );
+			return applyToManager( state, siteId, 'receive', true, posts, { query, found } );
 		},
 		[ POSTS_RECEIVE ]: ( state, { posts } ) => {
 			return reduce( groupBy( posts, 'site_ID' ), ( memo, sitePosts, siteId ) => {
-				return applyToManager( memo, siteId, 'receive', sitePosts );
+				return applyToManager( memo, siteId, 'receive', true, sitePosts );
```

Comment:
Change in behavior here: this will create a new `PostQueryManager` if one does not exist.  When would this happen?  Does the new behavior need a test?